### ISSUE TITLE
gelbooru.com

### ISSUE THREAD

**Issue by Aurani** [jspenguin2017/AdBlockProtector#182]
Not sure if this issue is because the site itself improved their anti-adblock script or because something broke in the last daily update to your script, regardless.
As of today (it worked fine about 3h ago) I'm unable to view any images on this site, disabling uBlockO makes the images view as normal, disabling your script does nothing however.
[Example](http://i.imgur.com/1cVSO4C.png)

And before you ask, no, I've not made any changes to my browser (Waterfox) or anything uBlock related.

**Comment by jspenguin2017** [comment]
Yes, looks like the website got a new ad block detection script. 
![image](https://cloud.githubusercontent.com/assets/7283682/25786986/6ad68c16-335a-11e7-942a-c65b276abfa3.png)

Last time I updated Script was this morning, this is my commit: https://github.com/jspenguin2017/AdBlockProtector/commit/31a4d12985a68a5684a2dd3a71d1481343e033e9
The new code will only run on `gaana.com` domain.

**Comment by okiehsch** [comment]
> As of today (it worked fine about 3h ago) I'm unable to view any images on this site, disabling uBlockO 

Update your `uBO-filter list`, then you should be able view the images again.
Fixed with https://github.com/uBlockOrigin/uAssets/commit/30d69ebf1c3000f8598debf009ac4ee30e0abb12

**Comment by jspenguin2017** [comment]
@okiehsch Yep, that works!

**Comment by uBlock-user** [comment]
The images can' be seen but the notice was n't removed.

Add -` gelbooru.com###nup`

**Comment by kowith337** [comment]
Hi, it's maybe not anti-adblock, but I've observed and found all thumbnail links are use redirected links, this change also affect cosmetic filtering that block `gelbooru.com/redirect.php?` hyperlinks.

The inline-script only triggered when they cannot found any value for `abvertDar` and then show **disable adblock** notice message, also re-show ads at footer.

**Comment by uBlock-user** [comment]
Just add - `a.noAccess("abvertDar");` that kills anti adblock script entirely.

**Comment by Aurani** [comment]
I can view the images fine now, but the script to edit tags is now broken despite my custom rules that previously enabled it.
`@@http://$script,domain=gelbooru.com`
`@@https://$script,domain=gelbooru.com`

**Comment by jspenguin2017** [comment]
@okiehsch A lot of site functionalities are now broken, the warning box is gone, but comments are no longer working. This filter is bad: 
![image](https://cloud.githubusercontent.com/assets/7283682/25808979/e845d40e-33c8-11e7-9b31-93762d1b30ff.png)

**Comment by lain566** [comment]
![ublock _ _registro_de_peticiones_de_red_-_google_c_2017-05-08_12-59-07](https://cloud.githubusercontent.com/assets/27441315/25809123/64c6fe66-340c-11e7-9a93-377884777947.png)

**Comment by jspenguin2017** [comment]
@lain566 Oh, that might be a filter from AAK List, lemme check

**Comment by jspenguin2017** [comment]
OK, I see why inline script is blocked... But we need to fix those `href="javascript:..."` and "onclick="..."`.

**Comment by uBlock-user** [comment]
The solution I provided in https://github.com/jspenguin2017/AdBlockProtector/issues/182#issuecomment-299854766 doesn't break anything and kills the anti adblock script.

**Comment by jspenguin2017** [comment]
@uBlock-user Yes, but then it serves you first party ads.

**Comment by lain566** [comment]
This is breaking the site

![ublock _ _registro_de_peticiones_de_red_-_google_c_2017-05-08_12-40-29](https://cloud.githubusercontent.com/assets/27441315/25809677/0a20d8a4-340e-11e7-8694-00cb224a0734.png)

**Comment by uBlock-user** [comment]
@jspenguin2017 Where ? I don't see any ?

**Comment by jspenguin2017** [comment]
@uBlock-user Did you `$badfilter`ed the inline script blocking rule?

**Comment by kowith337** [comment]
Can I still prove my test?
I've using my custom filterlist that targeted to this site so far and didn't get any issues until today because they start to replace thumbnail links to use redirect links instead of go to the post directly.

PS. I didn't have any accounts on Gelbooru to test between different of usage or function while still not lower your guard (uBlock, uMatrix, etc.)
At my side after [fixed hiding hyperlink](https://github.com/kowith337/Gelbolube/issues/3), the site still run fine and still block ads, even not have userscript installed.

**Comment by jspenguin2017** [comment]
@uBlock-user With uBO filter disabled, this is what I see: 

<details>
<summary>NSFW</summary>

![image](https://cloud.githubusercontent.com/assets/7283682/25811439/8c4eaa92-33d0-11e7-8c7e-c75175bdd414.png)


</details>

**Comment by uBlock-user** [comment]
Does blocking inline-script break anything ?

**Comment by jspenguin2017** [comment]
@kowith337 @uBlock-user You were blocking `redirect.php`, but that is not in EasyList nor any other filters. The site currently works at 80% with uBO filter, some "minor" functionalities like comments are broken. Looks like the event handler was not binded. 

![image](https://cloud.githubusercontent.com/assets/7283682/25811664/3ad5ad40-33d1-11e7-8313-c4abc03be80a.png)

**Comment by jspenguin2017** [comment]
To solve constant disagreements in our tests, we now have [Standard Configuration](https://github.com/jspenguin2017/AdBlockProtector/blob/master/Notes/Standard%20Configuration.MD). 
Also, because of constant breakage with FireFox, we'll, once again, drop the support for FireFox.

**Comment by uBlock-user** [comment]
Fine, make the call. I'm okay with comment section breaking as I'm not a commentator and can't spend time just for comment section, so I'm leaving this one to you.

**Comment by jspenguin2017** [comment]
@uBlock-user I rarely leave comments neither, but we can't call this fixed just because we don't use things that breaks.

**Comment by uBlock-user** [comment]
Ofcourse we can't call it fixed "technically", however I'm choosing lesser of two evils as I don't want another 200+ comments added to the issue tracker just in the name of discussion, neither I have the patience to do that again.

**Comment by okiehsch** [comment]
> @okiehsch A lot of site functionalities are now broken, the warning box is gone, but comments are no longer working.

@jspenguin2017 
"`ajax.googleapis.com` is blocked by EasyList, the comments would still be broken without `inline-script`."
https://github.com/uBlockOrigin/uAssets/commit/293a4c2b410d9525db3f17a23f53db8c64249f7b#commitcomment-22061018

**Comment by jspenguin2017** [comment]
@okiehsch So we will wait for EasyList to fix that? Also, some of these options on the side bar are not really working neither: 
![image](https://cloud.githubusercontent.com/assets/7283682/25825409/25c957ba-3400-11e7-8c74-dd38c2431666.png)
I'm thinking of extracting those event handlers and bind them using Script.

**Comment by okiehsch** [comment]
> So we will wait for EasyList to fix that?

gorhill added the `inline-script` block not me, so you really should ask him.
You can do that here: https://github.com/uBlockOrigin/uAssets/commit/293a4c2b410d9525db3f17a23f53db8c64249f7b#comments

**Comment by jspenguin2017** [comment]
@okiehsch 
Dump of result page: https://pastebin.com/wQkPE5Dn
Dump of image page: https://pastebin.com/9UkUaJ0z

So this is my new plan: 

* Override inline script filter
* Break identifiers `abvertDar` and `ExoLoader`.

**Comment by jspenguin2017** [comment]
@okiehsch EasyList blocks jQuery, that's fine because we can put in our own 😎 
```JavaScript
if (a.domCmp(["gelbooru.com"])) {
    a.win.$ = a.$;
    if (a.win.location.pathname === "/") {
        a.on("load", function () {
            a.$("div").each(function () {
                if (a.$(this).text() === "Have you first tried disabling your AdBlock?") {
                    a.$(this).empty();
                } else {
                    a.config.debugMode && a.out.log(a.$(this).text());
                }
            });
        });
    } else {
        a.noAccess("abvertDar");
        a.noAccess("ExoLoader");
    }
}
```
`||gelbooru.com^$inline-script,badfilter`

**Comment by jspenguin2017** [comment]
OK, everything works on my end 😄

**Comment by lain566** [comment]
can you delete this rule

`gelbooru.com##.noticeError`

![image_list__gelbooru_-_google_chrome_2017-05-08_23-47-34](https://cloud.githubusercontent.com/assets/27441315/25826927/1d394d5a-3449-11e7-949d-d4d0f292d5f2.png)

**Comment by jspenguin2017** [comment]
@lain566 Thanks. That is done.

**Comment by lain566** [comment]
**click here** not working   
`https://gelbooru.com/index.php?page=post&s=view&id=3680957`
![absurdres_breasts_dandon_fuga_eyebrows_green_eyes__2017-05-09_00-39-00](https://cloud.githubusercontent.com/assets/27441315/25828347/05325b64-3450-11e7-8e61-f676c56f287e.png)

**Comment by jspenguin2017** [comment]
![image](https://cloud.githubusercontent.com/assets/7283682/25828919/538ef672-3410-11e7-804d-7903953d950d.png)
OK, there are some other conflicting filters...

**Comment by jspenguin2017** [comment]
OK, let's `$badfilter` these as well...
![image](https://cloud.githubusercontent.com/assets/7283682/25828981/afcc2f04-3410-11e7-8b96-d259cfa75547.png)

**Comment by jspenguin2017** [comment]
What's the rule that disables all other element hiding rules? `$badfilter` somehow doesn't work. 
This works, but I don't want to use it...
`$("style:contains(:root)").remove();`

**Comment by okiehsch** [comment]
````
gelbooru.com#@#[style*="height:"][width]
gelbooru.com#@#[style*="width:"][height]
````
should work. I don'think there is a rule that disables all cosmetic filtering.
You can toggle off cosmetic filtering on a per site basis, that will also fix this issue.

**Comment by lain566** [comment]
`Paginator` and `Add Reply` is blocked by EasyList

https://gelbooru.com/index.php?page=forum&s=view&id=3437

![02](https://cloud.githubusercontent.com/assets/27441315/25829682/0223001a-3458-11e7-9bd3-0158766b1418.png)

```
gelbooru.com###paginator
gelbooru.com##center > a[href]
```

![ublock _ _registro_de_peticiones_de_red_-_google_c_2017-05-08_12-41-26](https://cloud.githubusercontent.com/assets/27441315/25829115/97cb8cb2-3454-11e7-8e6e-354a972c4768.png)
![ublock _ _registro_de_peticiones_de_red_-_google_c_2017-05-08_12-36-13](https://cloud.githubusercontent.com/assets/27441315/25829122/9f5c33b4-3454-11e7-88a0-184740613323.png)

**Comment by jspenguin2017** [comment]
Here are all the rules in EasyList that is specific for this domain: 
```
||gelbooru.com*/frontend*.js
||gelbooru.com/lk.php$subdocument
||gelbooru.com/poll.php$subdocument
||gelbooru.com/protech.php$subdocument
||gelbooru.com/x/

||twitch.tv/widgets/live_embed_player.swf$domain=gelbooru.com

@@||gelbooru.com/index.php?$xmlhttprequest
@@||gelbooru.com/public/$xmlhttprequest,domain=gelbooru.com
@@||gelbooru.com/script/application.js$script
@@||platform.twitter.com^$script,domain=gelbooru.com
|http://$script,xmlhttprequest,domain=gelbooru.com
|https://$script,xmlhttprequest,domain=gelbooru.com

||thedoujin.com^$domain=gelbooru.com

gelbooru.com##[-abp-properties='*: *;base64,']

gelbooru.com##[src^="http://syndication.exoclick.com/"] + * + * + [class]

gelbooru.com##center > a[href]

gelbooru.com###paginator
gelbooru.com##[height="250"]
gelbooru.com##[href*="ccbill."]
gelbooru.com##[href][rel*="nofollow"] video
gelbooru.com##[href][target*="blank"] video
gelbooru.com##[rel*="nofollow"] > [style]
gelbooru.com##[style*="height:"][width]
gelbooru.com##[style*="height:250px;"]
gelbooru.com##[style*="width:"][height]
gelbooru.com##a > [style][autoplay]
gelbooru.com##a[href*="/clicks.cgi"]
gelbooru.com##a[href*="?"][rel] > video
gelbooru.com##a[href] > :last-child > img[src]
gelbooru.com##a[id][href][rel*="nofollow"]
gelbooru.com##center > div > style + div[class]
gelbooru.com##center > div[style*="overflow:"]
gelbooru.com##div > [rel] > video[width]
gelbooru.com##div > div[id] > [href] > img[src]
gelbooru.com##div[style] > [class][style]
gelbooru.com##iframe + style + [class]
gelbooru.com##iframe[width]
gelbooru.com##span > style[type="text/css"] + *
gelbooru.com##span[class]:last-child > a img[src]

@@||ads.exoclick.com/*.js$script,domain=gelbooru.com|thepiratebay.org

@@||gelbooru.com//images/ad/$domain=gelbooru.com
```

**Comment by jspenguin2017** [comment]
So excluding rules that are already added to List, we need these? 
```
gelbooru.com#@#[style*="height:"][width]
gelbooru.com#@#[style*="width:"][height]
gelbooru.com#@##paginator
gelbooru.com#@#center > a[href]
```

**Comment by lain566** [comment]
@jspenguin2017 Yes, that works!

**Comment by jspenguin2017** [comment]
Looks like they just got some new first party ads, I added these to List, can someone test it out? 
```
gelbooru.com#@#[style*="height:"][width]
gelbooru.com#@#[style*="width:"][height]
gelbooru.com#@##paginator
gelbooru.com#@#center > a[href]
||assets.gelbooru.com/r19/
```

Also, for convenience, here are some test links: 
Search page: https://gelbooru.com/index.php?page=post&s=list&tags=breasts
Image page: https://gelbooru.com/index.php?page=post&s=view&id=3680957
Forum page: https://gelbooru.com/index.php?page=forum&s=view&id=3437

**Comment by lain566** [comment]
Everything works fine, but add this too

it's changing /r18/ /r19/...
`||assets.gelbooru.com/r18/`


block popups
`||gelbooru.com/script/application.js.php$important`


Doujin (thedoujin.com), friend site, this site has many ads
`||thedoujin.com/bload.php`

**Comment by jspenguin2017** [comment]
@lain566 Thanks, I'll test it out.

**Comment by jspenguin2017** [comment]
@lain566 Looks good, added to List.

**Comment by lain566** [comment]
@jspenguin2017 thanks

**Comment by jspenguin2017** [comment]
Looks like everything is still working. Closing. 
Please open another issue if it breaks again as this thread is pretty long.
